server/controllers/AutomationController.js

```diff
@@ -0,0 +1,283 @@
+    },
+    timeout: 1000,
+    httpAgent
+};
+
+const throwNoJobFoundError = jobId => {
+    throw new HttpQueryError(
+        404,
+        `Could not find job with jobId: ${jobId}`,
+        true
+    );
+};
+
+const throwNoTestFoundError = testCsvRow => {
+    throw new HttpQueryError(
+        404,
+        `Could not find test at CSV row number ${testCsvRow}`,
+        true
+    );
+};
+
+const throwSchedulerError = schedulerResponse => {
+    throw new HttpQueryError(
+        502,
+        `Response scheduler did not give a correct response: ${schedulerResponse}`,
+        false
+    );
+};
+
+const cancelJob = async (req, res) => {
+    const automationSchedulerResponse = await axios.post(
+        `${process.env.AUTOMATION_SCHEDULER_URL}/jobs/${req.params.jobID}/cancel`,
+        {},
+        axiosConfig
+    );
+
+    if (!automationSchedulerResponse.data) {
+        throwSchedulerError(automationSchedulerResponse);
+    }
+
+    if (
+        automationSchedulerResponse.data.status ===
+        COLLECTION_JOB_STATUS.CANCELLED
+    ) {
+        const graphqlRes = await updateCollectionJob(req.params.jobID, {
+            status: COLLECTION_JOB_STATUS.CANCELLED
+        });
+        if (!graphqlRes) {
+            throwNoJobFoundError(req.params.jobID);
+        }
+    }
+    res.json(automationSchedulerResponse.data);
+};
+
+const updateJobStatus = async (req, res) => {
+    const { status, externalLogsUrl } = req.body;
+
+    if (!Object.values(COLLECTION_JOB_STATUS).includes(status)) {
+        throw new HttpQueryError(400, `Invalid status: ${status}`, true);
+    }
+
+    const updatePayload = {
+        status,
+        ...(externalLogsUrl != null && { externalLogsUrl })
+    };
+
+    const graphqlResponse = await updateCollectionJob(
+        req.params.jobID,
+        updatePayload
+    );
+
+    if (!graphqlResponse) {
+        throwNoJobFoundError(req.params.jobID);
+    }
+
+    res.json(graphqlResponse);
+};
+
+const getApprovedFinalizedTestResults = async testPlanRun => {
+    const {
+        testPlanReport: { testPlanVersion }
+    } = testPlanRun;
+
+    // To be considered "Approved", a test plan run must be associated with a test plan report
+    // that is associated with a test plan version that is in "CANDIDATE" or "RECOMMENDED" or
+    // "DRAFT" phase and the test plan report been marked as final.
+    const { phase } = testPlanVersion;
+
+    if (
+        phase === 'RD' ||
+        (phase === 'DRAFT' && testPlanRun.testPlanReport.markedFinalAt === null)
+    ) {
+        return null;
+    }
+
+    const { testPlanReport } = await populateData({
+        testPlanReportId: testPlanRun.testPlanReport.id
+    });
+
+    return await getFinalizedTestResults({
+        ...testPlanReport
+    });
+};
+
+const updateOrCreateTestResultWithResponses = async ({
+    testCsvRow,
+    testPlanRun,
+    responses,
+    atVersionId,
+    browserVersionId
+}) => {
+    const allTestsForTestPlanVersion = await testsResolver(
```

Comment:
Should this use `getTests` from `server/models/services/TestsService` instead?

Reply:
Makes sense!